domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/routes/Route.java

```diff
@@ -0,0 +1,304 @@
+    }
+
+    @Nullable
+    public Integer getRouteSortOrder() {
+        return routeSortOrder;
+    }
+
+    /**
+     * Builder class to create {@link Route} objects. Allows an unordered definition of the different attributes of
+     * {@link Route}.
+     */
+    public static class RouteBuilder {
+
+        private String routeId;
+        private String agencyId;
+        private String routeShortName;
+        private String routeLongName;
+        private String routeDesc;
+        private RouteType routeType;
+        private String routeUrl;
+        private String routeColor;
+        private String routeTextColor;
+        private Integer routeSortOrder;
+
+        /**
+         * Sets field routeId value and returns this
+         *
+         * @param routeId identifies a route
+         * @return builder for future object creation
+         */
+        public RouteBuilder routeId(@NotNull final String routeId) {
+            this.routeId = routeId;
+            return this;
+        }
+
+        /**
+         * Sets field agencyId value and returns this
+         *
+         * @param agencyId agency for the specified route
+         * @return builder for future object creation
+         */
+        public RouteBuilder agencyId(@Nullable final String agencyId) {
+            this.agencyId = agencyId;
+            return this;
+        }
+
+        /**
+         * Sets field routeShortName value and returns this
+         *
+         * @param routeShortName short name of a route
+         * @return builder for future object creation
+         */
+        public RouteBuilder routeShortName(@Nullable final String routeShortName) {
+            this.routeShortName = routeShortName;
+            return this;
+        }
+
+        /**
+         * Sets field routeLongName value and returns this
+         *
+         * @param routeLongName full name of a route
+         * @return builder for future object creation
+         */
+        public RouteBuilder routeLongName(@Nullable final String routeLongName) {
+            this.routeLongName = routeLongName;
+            return this;
+        }
+
+        /**
+         * Sets field routeDesc value and returns this
+         *
+         * @param routeDesc Description of a route that provides useful, quality information.
+         * @return builder for future object creation
+         */
+        public RouteBuilder routeDesc(@Nullable final String routeDesc) {
+            this.routeDesc = routeDesc;
+            return this;
+        }
+
+        /**
+         * Sets field routeType value and returns this
+         *
+         * @param routeType Indicates the type of transportation used on a route
+         * @return builder for future object creation
+         */
+        public RouteBuilder routeType(final int routeType) throws NullPointerException {
+
+            try {
+                this.routeType = RouteType.fromInt(routeType);
+            } catch (NullPointerException ignored) {
```

Comment:
Would this be a correct way to implement the first solution you suggested? @barbeau 
``` 
        public RouteBuilder routeType(final int routeType) throws IllegalArgumentException {
            try {
                this.routeType = RouteType.fromInt(routeType);
            }
            finally{
                return this;
            }
        }
```

Reply:
In the implemented solution fromInt does not throw exception, instead it returns Null in case the value is unexpected. Then in the builder build method there is a null check that throws an IllegalArgumentException when encountering a null value for field routeType